src/astx/operators.py

```diff
@@ -182,3 +182,47 @@ def get_struct(self, simplified: bool = False) -> ReprStruct:
             "value": self.value.get_struct(simplified),
         }
         return self._prepare_struct(key, value, simplified)
+
+
+@public
+@typechecked
+class CompareOp(DataType):
+    """AST class for comparison operators acting as properties."""
+
+    def __init__(
+        self,
+        left: DataType,
+        ops: Iterable[Literal["==", "!=", "<", ">", "<=", ">="]],
+        comparators: Iterable[DataType],
+        loc: SourceLocation = NO_SOURCE_LOCATION,
+    ) -> None:
+        """Initialize the CompareOp instance."""
+        super().__init__(loc=loc)
+        self.ops = list(ops)
+        self.comparators = list(comparators)
+        if len(self.ops) != len(self.comparators):
+            raise ValueError(
+                "Number of operators must equal number of comparators."
+            )
+        for op in self.ops:
```

Comment:
this is not necessary because `@typechecked` will check the type, which is `Iterable[Literal["==", "!=", "<", ">", "<=", ">="]]`.